gatsby/createResolvers.mjs

```diff
@@ -0,0 +1,229 @@
+      type: 'TracksOrder'
+    });
+
+    entries = [...entries].sort((a, b) => {
+      let aIndex = trackOrder.indexOf(`${a.type}-tracks/${a.slug}`);
+      aIndex = aIndex > -1 ? aIndex : 9999;
+
+      let bIndex = trackOrder.indexOf(`${b.type}-tracks/${b.slug}`);
+      bIndex = bIndex > -1 ? bIndex : 9999;
+
+      return aIndex - bIndex;
+    });
+  }
+
+  return entries;
+};
+
+const filterByAuthorNameSlugResolver = async (args, context) => {
+  const { authorSlug, skip, limit } = args;
+
+  const query = {};
+
+  set(query, 'sort.fields', ['submittedOn', 'title']);
+  set(query, 'sort.order', ['DESC', 'ASC']);
+
+  if (authorSlug) set(query, `filter.author.nameSlug.eq`, authorSlug);
+  if (skip) set(query, 'skip', skip);
+  if (limit) set(query, 'limit', limit);
+
+  const { entries } = await context.nodeModel.findAll({
+    type: 'Contribution',
+    query
+  });
+
+  return entries;
+};
+
+const showcaseResolver = async (source, args, context, info) => {
+  const query = {};
+
+  set(query, 'filter.video.id.eq', source.id);
+  set(query, 'sort.order', ['ASC']);
+  set(query, 'sort.fields', ['name']);
+
+  const { entries } = await context.nodeModel.findAll({
+    type: 'Contribution',
+    query
+  });
+
+  return entries;
+};
+
+const videoCanonicalTrackResolver = async (source, args, context, info) => {
+  // we need a pointer from a video to its canonical track for the showcase page
+
+  // challenge videos can't have a canonical track (their challenge page is canon)
+  if (source.source === 'Challenges') return;
+
+  // a canonical track might've been explicitly set in the Gatsby node creation phase
+  if (source.canonicalTrack) {
+    return await context.nodeModel.getNodeById({ id: source.canonicalTrack });
+  }
+
+  // find tracks referencing this video
+  const queries = [
+    set({}, 'filter.videos.elemMatch.id.eq', source.id),
+    set({}, 'filter.chapters.elemMatch.videos.elemMatch.id.eq', source.id)
+  ];
+
+  for (const query of queries) {
+    const track = await context.nodeModel.findOne({ type: 'Track', query });
+    if (track) return track;
+  }
+
+  // it shouldn't be possible to get here - non-challenge videos should have at least one track referencing them
+  return;
+};
+
+const contributionSubmittedOnResolver = async (source, args, context, info) => {
+  // normalizes values to provide a stable sorting order for the showcase page
+
+  if (typeof source.submittedOn === 'string') {
+    // normalize to ISO 8601 since some dates are in `yyyy-mm-dd` format
+    return new Date(source.submittedOn).toISOString();
+  } else {
+    // the contribution doesn't have a `submittedOn` property, let's derive one for sorting purposes
+
+    // extract the sequential number from the source JSON filename
+    const [seqNumber] = source.name.match(/\d+/g);
+
+    // edge case where the sequential number is actually a timestamp yet `submittedOn` was not set
+    if (seqNumber.length > 6) return new Date(+seqNumber).toISOString();
```

Comment:
afaik this edge case shouldn't be necessary as all such videos already have their `submittedOn` set

Reply:
You're right, I verified and we don't have this edge case in the content anymore. I believe we did when the resolver was originally written but we can probably remove this. Thanks for the review!